### PR TITLE
Added GetDoubleNonZero Function

### PR DESCRIPTION
**Pull request by jgh9094**
I have added the GetDoubleNonZero function so that the doubles are between (0,1]. The original GetDouble function allowed for zeros to be returned, which causes issues in the GetRandNormal function, as log(0) = inf. 

The GetDoubleNonZero function rerolls until a nonzero number is returned.

### CODE REVIEW

**Comment by mmore500** [comment]
The test documentation build CI failure should now be resolved

File: third-party/Catch
Comment (mmore500) [inline third-party/Catch:1]: Do you remember why you needed to bump the Catch pin?

File: include/emp/math/Random.hpp [hunk 1]
```diff
@@ -382,10 +389,10 @@ namespace emp {
       // Using Rejection Method and saving of initial exponential random variable
       double expRV2;
       while (1) {
-        expRV2 = -log(GetDouble());
+        expRV2 = -log(GetDoubleNonZero());
```
Comment (mmore500) [inline include/emp/math/Random.hpp:392]: Yeah log-ing zero will not turn out well. Curious, did this actually happen in production? That sounds like the most unlucky bug ever (i.e., getting EXACTLY zero)!!!

File: include/emp/math/Random.hpp [hunk 2]
```diff
@@ -105,7 +105,14 @@ namespace emp {
     /// @return A pseudo-random double in the provided range.
     inline double GetDouble(const Range<double> range) noexcept {
       return GetDouble(range.GetLower(), range.GetUpper());
-     }
+    }
+
+    /// @return A pseudo-random double value between (0.0, 1.0]
+    inline double GetDoubleNonZero() noexcept {
+      double d = Get() / (double) RAND_CAP;
```
Comment (mmore500) [inline include/emp/math/Random.hpp:112]: Maybe delegate to existing implementation by calling GetDouble({std::numeric_limits::min(), 1.0}) ?

https://en.cppreference.com/w/cpp/types/numeric_limits/min

File: include/emp/bits/BitVector.hpp [hunk 3]
```diff
@@ -1558,7 +1558,7 @@ namespace emp {
   /// @return Read-only span of BitVector's bytes.
   std::span<const std::byte> BitVector::GetBytes() const {
     return std::span<const std::byte>(
-      bits.ReinterpretCast<const std::byte>(),
+      bits.ReinterpretCast<const std::byte>().Raw(),
```
Comment (mmore500) [inline include/emp/bits/BitVector.hpp:1561]: How is this related to adding the GetDoubleNonZero function?

**Review by mmore500 — COMMENTED** [review]
No problems, but a few questions!

File: include/emp/math/Random.hpp (hunk 1 above)
Comment (mmore500) [reply include/emp/math/Random.hpp:392]: 😅 omg that sounds horrible to debug. Glad you caught this :)

**Review by mercere99 — APPROVED** [review]
Changes are good (including fix to BitVector).  It's going to make the merger with MABE_devel more complicated due to incompatible changes, but they'll all still be in my head as long as I do that merge soon!  ;-)
